src/basic/chattr-util.c

```diff
@@ -172,3 +173,93 @@ int read_attr_at(int dir_fd, const char *path, unsigned *ret) {
 
         return read_attr_fd(fd, ret);
 }
+
+int read_xattr_flags_fd(int fd, unsigned *ret) {
+        struct fsxattr attrs;
+        _cleanup_close_ int fd_reopened = -EBADF;
+
+        assert(ret);
+
+        fd = fd_reopen_condition(fd, 0, O_PATH|O_CLOEXEC|O_NOCTTY|O_RDONLY, &fd_reopened);
```

Comment:
Please add short comment for `0`: `/* flags = */ 0`
But, the combination of flags and mask looks spurious.
I guess something like the following:
```
fd_reopen_condition(fd, O_RDONLY|O_CLOEXEC|O_NOCTTY, O_PATH, &fd_reopened);
```